### DISCUSSION TITLE
configuring daemon log levels from operator

### DISCUSSION THREAD

**Discussion by yuvalif** [rook/rook#9126] [Ideas]
currently, changing the daemons log level is done via the toolbox.
in the case of the RGW this is difficult because the user needs to know the client id of the RGW. 
e.g.
```
kubectl  - n rook-ceph exec - it deploy/rook-ceph-tools -- ceph config set client.rgw.my.store.a debug_rgw 10`
```
instead, it would be easier to set as a parameter in the cluster CR for osd, mon and mgr (could be per daemon as they already have per daemon sections in the CR). and could be in the object store CR for the rgw (under the gateway section).

please let me know what you think?

**Comment by leseb** [comment]
The RGW case makes sense, however, I have mixed feelings about this. I'm always hoping not to clutter the CRD too much. Changing the log level is not something the user has to do very often, so I'm sure if it's worth a CR spec addition at this point. If RGW is tricky we could simply solve this with some additional documentation. Also, I wonder if just applying `ceph config set client.rgw debug_rgw 10` would work... @yuvalif can you try? Thanks

**Reply by travisn** [reply]
Agreed on the concern of exposing this in the CRs. Changing the log level is usually only during troubleshooting where you can run a quick toolbox command to enable it, debug the issue, then change the logging back to the default. If there is a need to enable the logging permanently from cluster creation, another option is also the [config overrides](https://rook.github.io/docs/rook/v1.7/ceph-advanced-configuration.html#custom-cephconf-settings).

**Reply by yuvalif** [reply]
> The RGW case makes sense, however, I have mixed feelings about this. I'm always hoping not to clutter the CRD too much. Changing the log level is not something the user has to do very often, so I'm sure if it's worth a CR spec addition at this point. If RGW is tricky we could simply solve this with some additional documentation. Also, I wonder if just applying `ceph config set client.rgw debug_rgw 10` would work... @yuvalif can you try? Thanks

just tried that, it does not work. but setting config parameters for the RGW via the `ceph` tool is not in a good state anyway.
currently, the best option would be to do that via the admin socket, something like:
```
ceph daemon /var/run/ceph/ceph-client.rgw.ocs.storagecluster.cephobjectstore.b.1.94496229348720.asok config set config set debug_rgw 20
```

**Comment by Alvaro-Campesino** [comment]
I think this will be a good way of making easier and more understandable the logging levels, besides it can give more quick knowledge of the log level without having to get into the operator container.

**Comment by travisn** [comment]
Making the log levels simpler to set will be a helpful addition to the [Krew plugin](https://github.com/rook/kubectl-rook-ceph).
@yuvalif @Alvaro-Campesino Want to open an issue in that repo?

**Reply by yuvalif** [reply]
this sounds like a good idea. given the issue mentioned in: https://github.com/rook/rook/discussions/9126#discussioncomment-1624985
we should probably add some magic to that bash script, that in the case of the RGW, it will find the admin socket, and then run the command through it
